### PR TITLE
Reset Password API v2 endpoints

### PR DESCRIPTION
**Pull request by pmilewski**

### CODE REVIEW

File: app/controllers/spree/api/v2/storefront/passwords_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+          def update
+            user = Spree.user_class.reset_password_by_token(
+              password: params[:user][:password],
+              password_confirmation: params[:user][:password_confirmation],
+              reset_password_token: params[:id]
+            )
+
+            if user.errors.empty?
+              head :ok
```
Comment (chhga) [inline app/controllers/spree/api/v2/storefront/passwords_controller.rb:24]: should we just return head, not json?

Comment (pmilewski) [reply app/controllers/spree/api/v2/storefront/passwords_controller.rb:24]: > should we just return head, not json?

@damianlegawiec wdyt?

Comment (damianlegawiec) [reply app/controllers/spree/api/v2/storefront/passwords_controller.rb:24]: @chhga @pmilewski head is OK

File: app/controllers/spree/api/v2/storefront/passwords_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            else
+              head :unprocessable_entity
```
Comment (damianlegawiec) [inline app/controllers/spree/api/v2/storefront/passwords_controller.rb:26]: @pmilewski we need to return that error message

```suggestion
              json: { error: user.errors.full_messages.to_sentence }
```

**Pull request merged**